### DISCUSSION TITLE
Return Failed Verification instead of throwing a exception?

### DISCUSSION THREAD

**Discussion by Leonardo-Ferreira** [umbraco/Umbraco-CMS#15729] [Features and ideas]
why don't we create a new value for `PasswordVerificationResult` like `CouldNotExecute` or `MissingParameters` or even `UserCannotBeNull` instead of [throwing a new exception](https://github.com/umbraco/Umbraco-CMS/blob/8145a90094da67335306cd66ace3fdd8dae417d5/src/Umbraco.Infrastructure/Security/UmbracoPasswordHasher.cs#L45C13-L45C59)?

**Comment by bergmania** [comment]
I'm not sure I understand your request, but the method do not work if you have a `null` user. Further more `PasswordVerificationResult ` is a enum from `AspNetCore.Identity` and the method overrides one of their methods, so we should definitely stick to their contract.
